app/controllers/api/v1/medias_controller.rb

```diff
@@ -14,8 +14,13 @@ def index
         @url = params[:url]
         @refresh = params[:refresh] == '1'
         (render_parameters_missing and return) if @url.blank?
+        (render_error('The URL is not valid', 'INVALID_VALUE') and return) unless @url =~ /\A#{URI::regexp(%w(http https))}\z/
         @id = Digest::MD5.hexdigest(@url)
-        render_timeout { @media = Media.new(url: @url) } and return
+        begin
+          render_timeout { @media = Media.new(url: @url) } and return
+        rescue Exception => e
+          render_error("Could not parse URL (#{e.message})", 'UNKNOWN') and return
```

Comment:
If this exception is thrown, it means that the URL validation can't handle all cases? I would like to avoid rescuing generic `Exception` class